### ISSUE TITLE
Access the population during GA evolution

### ISSUE THREAD

**Issue by ropinotex** [wildart/Evolutionary.jl#122]
Is it possible to access the current population at the end of each iteration, for example by using the callbacks (or any other feasible way)?

Say, I want to check the population at the end of each iteration, and adding new individuals if some conditions are satisfied.

**Comment by jonathanfischer97** [comment]
Yes, but you would have define a `trace!` overload that saves a reference to your `state.population` in order to expose it to a callback for manipulation. The callback takes a trace record as input.
